lua/entities/gmod_tardis/modules/teleport/sh_tp_attached.lua

```diff
@@ -0,0 +1,106 @@
+if SERVER then
+    ENT:AddHook("DematStart", "attached", function(self)
+        local constrained = constraint.GetAllConstrainedEntities(self)
+        local attached
+        if constrained then
+            for k,v in pairs(constrained) do
+                if not (k.TardisPart or k==self) then
+                    local a=k:GetColor().a
+                    if not attached then attached = {} end
+                    attached[k] = a
+                end
+            end
+        end
+        self:SetData("demat-attached",attached,true)
+    end)
+
+    ENT:AddHook("StopDemat", "attached", function(self)
+        local attached=self:GetData("demat-attached")
+        if attached then
+            for k,v in pairs(attached) do
+                if IsValid(k) and not IsValid(k:GetParent()) then
+                    local phys=k:GetPhysicsObject()
+                    if phys and IsValid(phys) then
+                        k:SetSolid(SOLID_NONE)
+                        k.gravity = phys:IsGravityEnabled()
+                        phys:EnableGravity(false)
+                    end
+                end
+            end
+        end
+    end)
+
+    ENT:AddHook("StopMat", "attached", function(self)
+        local attached=self:GetData("demat-attached")
+        if attached then
+            for k,v in pairs(attached) do
+                if IsValid(k) then
+                    k:SetColor(ColorAlpha(k:GetColor(),v))
+                end
+            end
+        end
+        self:SetData("demat-attached",nil,true)
+    end)
+
+    ENT:AddHook("PreTeleportPositionChange", "attached_hoverballs", function(self, pos, ang, phys_enable)
+        local attached=self:GetData("demat-attached")
+        if attached then
+            for k,v in pairs(attached) do
+                if IsValid(k) and not IsValid(k:GetParent()) then
+                    k.telepos=k:GetPos()-self:GetPos()
+                    if k:GetClass()=="gmod_hoverball" then -- fixes hoverballs spazzing out
```

Comment:
Been a long time since I wrote this comment 😅 the wording should change here to be inoffensive now. Maybe 'fixes hoverballs freaking out'?

Reply:
After discussion this is fine actually, the term "spazzing out" refers to things like muscle spasms and I think that is clear enough given the context here